---
title: "Bis15L Final Project"
author: "Omar Afzal"
date: "`r Sys.Date()`"
output:
  html_document: 
    keep_md: yes
    theme: spacelab
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


## Load the libraries
```{r}
library(tidyverse)
library(janitor)
library(here)
library(ggthemes)
if (!require("qqman")) install.packages('qqman')
library(qqman)
library(dplyr)
if (!require("ggrepel")) install.packages('ggrepel')
library(ggrepel)
```


```{r}
#What are SNPs and why do they matter? Cancer is a conglomerate of gain-of-function of oncogenes and the loss-of-function of tumor suppressor genes. Identification of single-nucleotide polymorphisms and other mutations at important gene sites can be correlated with being cancer drivers in addition to their disease rates and severity. 
```

```{r}
#How do we interpret odds ratios and GWAS statistics?

#What are the odds of developing breast cancer if my Chromosome 5 - position 58804536 (rs4326095 SNP) alleles are "GA"?
```
$$
\textrm{odds if GA} = \frac{\textrm{breast cancer|GA}}{\textrm{no breast cancer|GA}} 
$$
$$
\textrm{odds if GC or AT} = \frac{\textrm{breast cancer|GC or AT}}{\textrm{no breast cancer|GC or AT}}
$$
$$
\textrm{odds ratio} = \frac{\textrm{odds if GA}}{\textrm{odds if GC or AT}} = 1.4084 
$$

```{r}
SNP_data <- readr::read_csv("data/SNPv2.csv")
SNP_data <- janitor::clean_names(SNP_data)
glimpse(SNP_data) #Association between a single SNP and breast cancer susceptibility in a group of 1,145 breast cancer patients and 1,142 controls.
SNP_data %>%
  arrange(desc(odds_ratio))

```


```{r}
SNP_data %>%
  group_by(chr_id) %>%
  summarise(n_distinct(snp_id))
```

```{r}
SNP_data %>%
  ggplot(aes(x = chr_id, y=n_distinct(snp_id))) +
  labs(title = "Distribution of SNPs",
       x = "Chromosome",
       y = "SNP count")+ theme_stata()+
  theme(axis.text.x = element_text(hjust = 1))+
  geom_col(fill="green",color="black")+
  scale_x_continuous(breaks = seq(1, 21, by = 1))
SNP_data %>% 
  ggplot(aes(x=chr_id, fill=n_distinct(snp_id))) + geom_density(alpha=.4)
```


```{r}
SNP_data_causal <- SNP_data %>%
  filter(p_value<0.1)%>% #p-value from HWE testing in cases of breast cancer
  group_by(chr_id) %>%
  summarise(n_distinct(snp_id))
  
```

```{r}
SNP_data_causal %>%
  ggplot(aes(x = chr_id, y=`n_distinct(snp_id)`)) +
  labs(title = "Distribution of SNPs with p-value <0.1",
       x = "Chromosome",
       y = "SNP count")+ theme_stata()+
  theme(axis.text.x = element_text(hjust = 1))+
  geom_col(fill="pink",color="black")
 
```

```{r}
### MANHATTAN GRAPH - too advanced to draw by hand ###
don <- gwasResults %>% 
  
  # Compute chromosome size
  group_by(CHR) %>% 
  summarise(chr_len=max(BP)) %>% 
  
  # Calculate cumulative position of each chromosome
  mutate(tot=cumsum(chr_len)-chr_len) %>%
  select(-chr_len) %>%
  
  # Add this info to the initial dataset
  left_join(gwasResults, ., by=c("CHR"="CHR")) %>%
  
  # Add a cumulative position of each SNP
  arrange(CHR, BP) %>%
  mutate( Chromosome=BP+tot) %>%

  # Add highlight and annotation information
  mutate( is_highlight=ifelse(SNP %in% snpsOfInterest, "yes", "no")) %>%
  mutate( is_annotate=ifelse(-log10(P)>4, "yes", "no")) 

# Prepare X axis
axisdf <- don %>% group_by(CHR) %>% summarize(center=( max(Chromosome) + min(Chromosome) ) / 2 )

# Make the plot
ggplot(don, aes(x=Chromosome, y=-log10(P))) +
    
    # Show all points
    geom_point( aes(color=as.factor(CHR)), alpha=0.8, size=1.3) +
    scale_color_manual(values = rep(c("grey", "skyblue"), 22 )) +
    
    # custom X axis:
    scale_x_continuous( label = axisdf$CHR, breaks= axisdf$center ) +
    scale_y_continuous(expand = c(0, 0) ) +     # remove space between plot area and x axis

    # Add highlighted points
    geom_point(data=subset(don, is_highlight=="yes"), color="orange", size=2) +

    # Custom the theme:
    theme_bw() +
    theme( 
      legend.position="none",
      panel.border = element_blank(),
      panel.grid.major.x = element_blank(),
      panel.grid.minor.x = element_blank()
    )
```